src/Modules/DatePickerInput.react.js

```diff
@@ -310,46 +294,66 @@ class DatePickerInput extends Component {
         }
     }
 
-    _getAllowedDateFormats(specifiedFormat) {
-        const formats = DateUtils.getAllowedDateFormats();
-
-        if (_.indexOf(formats, specifiedFormat) < 0) {
-            formats.unshift(specifiedFormat);
-        }
-
-        return formats;
+    _setOpen(open) {
+        this.setState({ isCalendarOpen: open });
     }
 }
 
+DatePickerInput.defaultProps = {
+    className: '',
+    controls: 'dropdowns',
+    date: undefined,
+    dateFrom: undefined,
+    dateTo: undefined,
+    disabled: false,
+    disabledDays: undefined,
+    events: undefined,
+    errorMessage: '',
+    excludeDates: undefined,
+    filterDates: undefined,
+    id: undefined,
+    includeDates: undefined,
+    label: '',
+    locale: '',
+    maxDate: undefined,
+    minDate: undefined,
+    onBlur: undefined,
+    onChange: undefined,
+    onFocus: undefined,
+    onMonthChange: undefined,
+    rangeFrom: false,
+    rangeTo: false,
+    required: false,
+    tabIndex: undefined,
+};
```

Comment:
Is it necessary to explicitly default props to `undefined`?  What do they default to otherwise if they're not specified here and not specified when used?

Reply:
I removed them. It was something new I was trying, but everything seemed to pretty much nil.

Reply:
Fair enough.  AFAIK `undefined` is the ultimate default in JavaScript ... you should never have to set it explicitly, unless you're trying to clear something out.  I tend to prefer `null` myself for a placeholder default non-value for something until it gets assigned a value, but likely that's the back-end dude in me (which of course wonders why we even have `undefined` in the first place  ... just force everyone to define their variables for crying out loud and fail hard if they don't ... enough with this `undefined` nonsense! 😁)